task_processing/plugins/mesos/mesos_executor.py

```diff
@@ -33,15 +30,17 @@ def __init__(
         Constructs the instance of a task execution, encapsulating all state
         required to run, monitor and stop the job.
 
-        :param dict credentials: Mesos principal and secret.
+        TODO param docstrings
         """
 
         self.logger = logging.getLogger(__name__)
+        self.role = role
 
         self.execution_framework = ExecutionFramework(
             role=role,
             pool=pool,
             name=framework_name,
+            handle_offer_callback=self.handle_offer,
```

Comment:
> This is not great. Is the only method that we need to pass to the execution framework?

Yea, I've gone back and forth on this a bunch.  I think we're going to need to have a couple other callbacks, one for handling Mesos events and the other for handling killing of tasks/pods.  The other option which we could do is just have a list of callbacks that a `MesosExecutor` is required to implement and then we could just pass the whole `MesosExecutor` object to the execution framework.

@Rob-Johnson what are your thoughts?

Reply:
It looks like the current approach is what we decided upon two weeks ago. In the interest of getting this shipped ASAP (and since we've already allowed extensive time for discussing and reviewing the proposed implementation) - let's just go with it for now.